ui/app/components/app/whats-new-popup/whats-new-popup.js

```diff
@@ -0,0 +1,174 @@
+import React, { useContext, useMemo, useRef, useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { I18nContext } from '../../../contexts/i18n';
+import { useEqualityCheck } from '../../../hooks/useEqualityCheck';
+import Button from '../../ui/button';
+import Popover from '../../ui/popover';
+import { updateViewedNotifications } from '../../../store/actions';
+import { getTranslatedUINoficiations } from '../../../../../shared/notifications';
+import { getSortedNotificationsToShow } from '../../../selectors';
+
+function getActionFunctions() {
+  const actionFunctions = {
+    1: () => {
+      global.platform.openTab({
+        url: 'https://metamask.io/download.html',
+      });
+    },
+    2: () => {
+      global.platform.openTab({
+        url:
+          'https://survey.alchemer.com/s3/6173069/MetaMask-Extension-NPS-January-2021',
+      });
+    },
+    3: () => {
+      global.platform.openTab({
+        url: 'https://community.metamask.io/t/about-the-security-category/72',
+      });
+    },
+  };
+
+  return actionFunctions;
+}
+
+export default function WhatsNewPopup({ onClose }) {
+  const t = useContext(I18nContext);
+
+  const actionFunctions = getActionFunctions();
+
+  const notifications = useSelector(getSortedNotificationsToShow);
+
+  const containerRef = useRef();
+
+  const memoizedNotifications = useEqualityCheck(notifications);
+  const idRefMap = useMemo(
+    () =>
+      memoizedNotifications.reduce(
+        (_idRefMap, notification) => ({
+          ..._idRefMap,
+          [notification.id]: React.createRef(),
+        }),
+        {},
+      ),
+    [memoizedNotifications],
+  );
+
+  const [seenNotifications, setSeenNotifications] = useState({});
+  const observationsCreated = useRef(false);
+
+  useEffect(() => {
+    const observers = [];
+    observationsCreated.current = true;
+
+    Object.entries(idRefMap).forEach(([id, ref]) => {
+      const observer = new window.IntersectionObserver(
```

Comment:
Why create a new observer for each element? We only need one intersection observer for each content area. It can be shared to observe all of the individual notifications.

Reply:
We need a way to identify the notification being observed in the callback passed to `IntersectionObserver`. One way is via closures: create an observer for each notification so that each callback uses an `id` for a specific notification

Reply:
Alternatively, the `entry` passed to the callback does contain a `target`, which I suppose we could get an id/key from and use that to identify which notifications have been seen.

Reply:
Exactly - the `target` property is how we're supposed to identify what the entry is for. We can use `Object.entries(idRefMap).find(() => ...)` to get the corresponding ID`

Reply:
@Gudahtt The logic was updated as per these comments in 233b70e5e

You'll notice the update involved more than just comparing refs from the idRefMap to the entry. While making this change I discovered that the asynchronous nature of the loading of the image (associated with the first notification) could mess with the intersection observer.

In particular, multiple notification divs were being observed as "intersecting" on first render but before the image loaded. However, they should not have been considered "seen" / observed because after the image loads they are out of view. The fix was to only create the observer once the image was loaded

Reply:
Huh, that makes sense, great catch!

One simpler alternative to watching for the image load event would be to set the `width` and `height` attributes of the `img` element. This would also prevent a re-flow as the image loads, resulting in a smoother loading experience.

MDN seems to recommend it:
>Use both width and height to set the intrinsic size of the image, allowing it to take up space before it loads, to mitigate content layout shifts.

It would necessitate a change in the notification data structure to accommodate image metadata (e.g. maybe the image path would get replaced with an image object), _and_ we'd risk the image metadata not matching the image itself (I'm not sure the easiest way to prevent this with automated lints/tests :thinking: ). But it's easy to verify manually, and at least it wouldn't require a change to the notification controller.

Reply:
Good suggestion.

Done in e805cf2fc